### ISSUE TITLE
how to make classification for seq task?

### ISSUE THREAD

**Issue by MohamedLotfyElrefai** [calofmijuck/pytorch-bert-fine-tuning#1]
you have said 
```
BERT model for classification.
This module is composed of the BERT model with a linear layer on top of
the pooled output.
```
but in your implementation of the network you aren't making a linear layer on the pooled layer to extract the feature label:
```
 self.bert = BertModel(config)
self.dropout = nn.Dropout(config.hidden_dropout_prob) ##

```
another question which i wanna understand as i loading a bert model from pytorch and after that adding a layer of classify to classify multiple labels
but how i will save those weights of the last layer as im doing only add in the forward model

this is the implementation:
```
from torch.nn import CrossEntropyLoss

class BertForSequenceClassification():
    """BERT model for classification.
    This module is composed of the BERT model with a linear layer on top of
    the pooled output.

    Params:
        `config`: a BertConfig class instance with the configuration to build a new model.
        `num_labels`: the number of classes for the classifier. Default = 2.

    Inputs:
        `input_ids`: a torch.LongTensor of shape [batch_size, sequence_length]
            with the word token indices in the vocabulary(see the tokens preprocessing logic in the scripts
            `extract_features.py`, `run_classifier.py` and `run_squad.py`)
        `token_type_ids`: an optional torch.LongTensor of shape [batch_size, sequence_length] with the token
            types indices selected in [0, 1]. Type 0 corresponds to a `sentence A` and type 1 corresponds to
            a `sentence B` token (see BERT paper for more details).
        `attention_mask`: an optional torch.LongTensor of shape [batch_size, sequence_length] with indices
            selected in [0, 1]. It's a mask to be used if the input sequence length is smaller than the max
            input sequence length in the current batch. It's the mask that we typically use for attention when
            a batch has varying length sentences.
        `labels`: labels for the classification output: torch.LongTensor of shape [batch_size]
            with indices selected in [0, ..., num_labels].

    Outputs:
        if `labels` is not `None`:
            Outputs the CrossEntropy classification loss of the output with the labels.
        if `labels` is `None`:
            Outputs the classification logits of shape [batch_size, num_labels].

    Example usage:
    ```python
    # Already been converted into WordPiece token ids
    input_ids = torch.LongTensor([[31, 51, 99], [15, 5, 0]])
    input_mask = torch.LongTensor([[1, 1, 1], [1, 1, 0]])
    token_type_ids = torch.LongTensor([[0, 0, 1], [0, 1, 0]])

    config = BertConfig(vocab_size_or_config_json_file=32000, hidden_size=768,
        num_hidden_layers=12, num_attention_heads=12, intermediate_size=3072)

    num_labels = 2

    model = BertForSequenceClassification(config, num_labels)
    logits = model(input_ids, token_type_ids, input_mask)
    ```
    """
    def __init__(self, path,do_train, num_labels=2):
        super(BertForSequenceClassification, self).__init__()
        self.num_labels = num_labels
        self.bert =  torch.load(path)
        self.model=nn.Sequential(self.bert , nn.Linear(60, num_labels))
        if do_train==True:
            self.bert.train()
        else:
            self.bert.val()
 #       self.dropout = nn.Dropout(config.hidden_dropout_prob)
        self.classifier = nn.Linear(60, num_labels)
      #  self.apply(self.init_bert_weights)
    def forward(self, input_ids, segment_ids, labels=None):
#first hidden units output from my bert model
        pooled_output = self.model.forward(torch.tensor(input_ids),torch.tensor(segment_ids))[:, 0]
        #pooled_output = self.dropout(pooled_output)
        logits = self.classifier(pooled_output)
        if labels is not None:
            loss_fct = CrossEntropyLoss()
            loss = loss_fct(logits.view(-1, self.num_labels), labels.view(-1))
            return loss
        else:
            return logits 
```
